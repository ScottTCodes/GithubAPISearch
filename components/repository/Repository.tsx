import React from 'react'
import Image from 'next/image'

import styles from '../../styles/Repository.module.css'

export interface Props {
  name: string,
  author: string,
  stars: string,
  image: string,
}

const Repository = ({ name, author, stars, image }: Props) => (
  <article className={styles.repository}>
    <div className={styles['repository__inner-wrap']}>
      <Image className={styles.repository__image} src={image} height={100} width={100} alt={name}  data-testid="image" />
      <div className={styles['repository__inner-wrap-content']}>
        <h2 className={styles.repository__name} data-testid="name">{name}</h2>
        <h3 className={styles.repository__author} data-testid="author">{author}</h3>
      </div>
    </div>
    <div className={styles.repository__details}>
      <p className={styles['repository__details-stat']}>
        <Image src="/stars.svg" height={30} width={30} alt="Star" />
        <span data-testid="stars">{stars} Stars</span>
      </p>
    </div>
  </article>
)

export default Repository
