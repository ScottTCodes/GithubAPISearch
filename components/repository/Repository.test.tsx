import React from "react"
import {
  render,
  screen,
} from '@testing-library/react'
import Repository, { IRepository } from './index'

const renderRepository = (props: Partial<IRepository> = {
    name: 'Arctic Shores Repository',
    author: 'ScottTCodes',
    stars: '69',
    watchers: '420',
    image: 'https://images.unsplash.com/photo-1506744038136-46273834b3fb?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxleHBsb3JlLWZlZWR8MXx8fGVufDB8fHx8&w=1000&q=80',
  }) => {
    const defaultProps: IRepository = {
      name: 'Arctic Shores Repository',
      author: 'ScottTCodes',
      stars: '69',
      watchers: '420',
      image: 'https://images.unsplash.com/photo-1506744038136-46273834b3fb?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxleHBsb3JlLWZlZWR8MXx8fGVufDB8fHx8&w=1000&q=80',
    };

    return render(<Repository {...defaultProps} {...props} />)
}

describe("<Repository />", () => {
  test("should render data from API", async () => {
    const { asFragment } = renderRepository();

    expect(asFragment()).toMatchSnapshot();
  });
})
