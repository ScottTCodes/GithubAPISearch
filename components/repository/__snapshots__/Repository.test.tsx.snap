// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repository /> should render data from API 1`] = `
<DocumentFragment>
  <article
    class="repository"
  >
    <div
      class="repository__inner-wrap"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg xmlns=%27http://www.w3.org/2000/svg%27 version=%271.1%27 width=%27100%27 height=%27100%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Arctic Shores Repository"
          class="repository__image"
          data-nimg="intrinsic"
          data-testid="image"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="repository__inner-wrap-content"
      >
        <h2
          class="repository__name"
          data-testid="name"
        >
          Arctic Shores Repository
        </h2>
        <h3
          class="repository__author"
          data-testid="author"
        >
          ScottTCodes
        </h3>
      </div>
    </div>
    <div
      class="repository__details"
    >
      <p
        class="repository__details-stat"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg xmlns=%27http://www.w3.org/2000/svg%27 version=%271.1%27 width=%2730%27 height=%2730%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Star"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <span
          data-testid="stars"
        >
          69 Stars
        </span>
      </p>
    </div>
  </article>
</DocumentFragment>
`;
